src/shogun/kernel/AUCKernel.cpp

```diff
@@ -1,121 +1,123 @@
 /*
  * This software is distributed under BSD 3-clause license (see LICENSE file).
  *
- * Authors: Soeren Sonnenburg, Evan Shelhamer, Chiyuan Zhang, Bjoern Esser, 
+ * Authors: Soeren Sonnenburg, Evan Shelhamer, Chiyuan Zhang, Bjoern Esser,
  *          Sergey Lisitsyn
  */
 
-#include <shogun/lib/common.h>
-#include <shogun/mathematics/Math.h>
-#include <shogun/kernel/AUCKernel.h>
 #include <shogun/io/SGIO.h>
+#include <shogun/kernel/AUCKernel.h>
 #include <shogun/labels/BinaryLabels.h>
+#include <shogun/lib/common.h>
+#include <shogun/mathematics/Math.h>
 
 using namespace shogun;
 
-void
-CAUCKernel::init()
+void CAUCKernel::init()
 {
-	SG_ADD((CSGObject**) &subkernel, "subkernel", "The subkernel.",
-	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &subkernel, "subkernel", "The subkernel.", ParameterProperties::HYPER);
+	SG_ADD(&labels, "labels", "The labels.");
+	watch_method("setup_auc_maximization", &CAUCKernel::setup_auc_maximization);
 }
 
-CAUCKernel::CAUCKernel()
-: CDotKernel(0), subkernel(NULL)
+CAUCKernel::CAUCKernel() : CDotKernel(0), subkernel(nullptr), labels(nullptr)
 {
 	init();
 }
 
-CAUCKernel::CAUCKernel(int32_t size, CKernel* s)
-: CDotKernel(size), subkernel(s)
+CAUCKernel::CAUCKernel(int32_t size, CKernel* s, CLabels* l)
+    : CDotKernel(size), subkernel(s), labels(l)
 {
 	init();
 	SG_REF(subkernel);
+	SG_REF(labels)
 }
 
 CAUCKernel::~CAUCKernel()
 {
 	SG_UNREF(subkernel);
+	SG_UNREF(labels)
 	cleanup();
 }
 
-CLabels* CAUCKernel::setup_auc_maximization(CLabels* labels)
+bool CAUCKernel::setup_auc_maximization()
```

Comment:
Not sure I understood this earlier... why does this return a book? Ie what happens with the return value when calling run?